ostd/src/mm/kspace/kva.rs

```diff
@@ -0,0 +1,420 @@
+                        if !flusher.need_remote_flush() && tlb_prefer_flush_all {
+                            // Only on single-CPU cases we can drop the page immediately before flushing.
+                            drop(page);
+                            continue;
+                        }
+                        flusher.issue_tlb_flush_with(TlbFlushOp::Address(va), page);
+                    }
+                    id if id == TypeId::of::<Untracked>() => {
+                        panic!("Found tracked memory mapped into untracked `Kva`");
+                    }
+                    _ => panic!("Unexpected `Kva` type"),
+                },
+                PageTableItem::MappedUntracked { va, .. } => match TypeId::of::<M>() {
+                    id if id == TypeId::of::<Untracked>() => {
+                        if !flusher.need_remote_flush() && tlb_prefer_flush_all {
+                            continue;
+                        }
+                        flusher.issue_tlb_flush(TlbFlushOp::Address(va));
+                    }
+                    id if id == TypeId::of::<Tracked>() => {
+                        panic!("Found untracked memory mapped into tracked `Kva`");
+                    }
+                    _ => panic!("Unexpected `Kva` type"),
+                },
+                PageTableItem::NotMapped { .. } => {
+                    break;
+                }
+                PageTableItem::PageTableNode { .. } => {
+                    panic!("Found page table node in `Kva`");
+                }
+            }
+        }
+
+        if !flusher.need_remote_flush() && tlb_prefer_flush_all {
+            flusher.issue_tlb_flush(TlbFlushOp::All);
+        }
+
+        flusher.dispatch_tlb_flush();
+
+        // 2. free the virtual block
+        let allocator = M::select_allocator();
+        allocator.free(range);
+    }
+}
+
+impl Kva<Untracked> {
+    fn reader(&self) -> VmReader<'_, Infallible> {
+        // SAFETY: The constructor of the `Kva<Untracked>` structure has already ensured the
+        // safety of reading from the mapped physical address, and the mapping is valid.
+        unsafe { VmReader::from_kernel_space(self.range.start as *mut u8, self.range.len()) }
+    }
+
+    fn writer(&self) -> VmWriter<'_, Infallible> {
+        // SAFETY: The constructor of the `Kva<Untracked>` structure has already ensured the
+        // safety of writing to the mapped physical address, and the mapping is valid.
+        unsafe { VmWriter::from_kernel_space(self.range.start as *mut u8, self.range.len()) }
+    }
+}
+
+impl VmIo for Kva<Untracked> {
+    fn read(&self, offset: usize, writer: &mut VmWriter) -> Result<()> {
+        if self
+            .range
+            .len()
+            .checked_sub(offset)
+            .is_none_or(|remain| remain < writer.avail())
+        {
+            return Err(Error::InvalidArgs);
+        }
+
+        // SAFETY: Reading from untracked memory is safe.
```

Comment:
There's no guarantee that reading from or writing to untracked-mapped memory is safe. It's often not safe. There's no unsafe code either. So you need to just remove these comments.

The reason why `VmIo` can be implemented for `Kva<Untracked>` is that the builder of `Kva<Untracked>` should ensure that the memory mapped is available for reading/writing.

Reply:
Forgot to remove the comments :<